src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.ts

```diff
@@ -75,7 +75,8 @@ export class CourseConsumptionHeaderComponent implements OnInit, AfterViewInit,
     if (!this.courseConsumptionService.getCoursePagePreviousUrl) {
       this.courseConsumptionService.setCoursePagePreviousUrl();
     }
-    this.isTrackable = _.lowerCase(_.get(this.courseHierarchy, 'trackable.enabled')) === 'yes';
+    this.isTrackable = _.lowerCase(_.get(this.courseHierarchy, 'trackable.enabled')) === 'yes' ||
```

Comment:
Why don't you put `isTrackable` inside the interface call & start calling the interface method for any course/collection.
Initialise the call with hierarchy data, & then call the interface methods.

This will helps to write this check in every component.